### ISSUE TITLE
How can I get port and proto in my action script ?

### ISSUE THREAD

**Issue by wiseflat** [fail2ban/fail2ban#1935]
### Environment:

- Fail2Ban version (including any possible distribution suffixes): 0.9.6
- OS, including release name/version: Ubuntu 16.04
- [ x] Fail2Ban installed via OS/distribution mechanisms
- [ x] You have not applied any additional foreign patches to the codebase
- [x ] Some customizations were done to the configuration (provide details below is so)

### The issue:

It's not an issue but a question. I would like to get the port and the protocol in my action script, but the port should not be specified in the jail.local specification. When I try the sample configuration above, I don't get the right value. Do you know how can I get them ?
Cheers

#### Any customizations done to /etc/fail2ban/ configuration
```
## /etc/fail2ban/jail.local

[shorewall]
enabled = true
port = 0:65535
filter = shorewall
logpath = /var/log/shorewall.log
maxretry = 1
```

```
## /etc/fail2ban/action.d/myaction.conf
actionban = echo ip=<ip> port=<port> proto=<proto> >> /tmp/test.log
```

**Comment by sebres** [comment]
You should provide this to the action as parameters, like:
```
action = myaction[name=%(__name__)s, bantime="%(bantime)s", port="%(port)s", proto="%(protocol)s", ...]
```
See [jail.conf](https://github.com/fail2ban/fail2ban/blob/0.9/config/jail.conf#L164) for example.

Please note, that for fail2ban version before v.0.10 you should necessarily define all this parameters in the section `Init` of the action.

**Comment by wiseflat** [comment]
Thank you for your answer. But It is not solved :-)
If I use this in my jail.local:
```
port = 0:65535
``` 

This is what I get on the log file :

```
ip=xxxx port=0:65535 proto=tcp
ip=xxxy port=0:65535 proto=tcp
ip=xxxz port=0:65535 proto=tcp
```

I would like to get automatically the port and the protocol based on the reject tcp/udp connexion from shorewall.log. As you can see, it fulfill automatically port value with : 0:65535, which is the value I wrote on my jail.local rule.

**Comment by sebres** [comment]
> I would like to get automatically the port and the protocol based on the reject tcp/udp connexion from shorewall.log.

This is:
- impossible with fail2ban v.0.9 at all
- possible since fail2ban v.0.10 **only** with custom filter (resp. a bit rewritten `failregex`)...
This was firstly introduced in #1698 (closes RFE #1110) as "interpolation of tags `<F-...>`".

No idea how the shorewall-log looks, so below an example for test.log:
```
Oct 24 21:56:13 failure from 1.2.3.4, port: 54568, protocol: tcp - user unknown
```

```INI
# jail.local or filter.d/myfilter.conf
failregex = ^failure from <HOST>, port: <F-PORT>\d+</F-PORT>, protocol: <F-PROTO>\S+</F-PROTO> - <F-REASON>.*</F-REASON>$

# myaction.conf:
actionban = echo ip=<ip> port=<F-PORT> proto=<F-PROTO> reason="<F-REASON>" >> /tmp/test.log
```

Hope this helps.
